app/workers/builds.php

```diff
@@ -248,22 +249,25 @@ protected function buildDeployment(GitHub $github, Document $project, Document $
                     );
                 }
 
-                Console::execute('tar --exclude code.tar.gz -czf /tmp/builds/' . \escapeshellcmd($buildId) . '/code.tar.gz -C /tmp/builds/' . \escapeshellcmd($buildId) . '/code' . (empty($rootDirectory) ? '' : '/' . $rootDirectory) . ' .', '', $stdout, $stderr);
+                $tmpPath = '/tmp/builds/' . \escapeshellcmd($buildId);
+                $tmpPathFile = $tmpPath . '/code.tar.gz';
 
-                $deviceFunctions = $this->getFunctionsDevice($project->getId());
+                Console::execute('tar --exclude code.tar.gz -czf ' . $tmpPathFile . ' -C /tmp/builds/' . \escapeshellcmd($buildId) . '/code' . (empty($rootDirectory) ? '' : '/' . $rootDirectory) . ' .', '', $stdout, $stderr);
 
-                $fileName = 'code.tar.gz';
-                $fileTmpName = '/tmp/builds/' . $buildId . '/code.tar.gz';
+                $deviceFunctions = $this->getFunctionsDevice($project->getId());
 
-                $path = $deviceFunctions->getPath($deployment->getId() . '.' . \pathinfo($fileName, PATHINFO_EXTENSION));
+                $localDevice = new Local();
+                $buffer = $localDevice->read($tmpPathFile);
+                $mimeType = $localDevice->getFileMimeType($tmpPathFile);
 
-                $result = $deviceFunctions->move($fileTmpName, $path);
+                $path = $deviceFunctions->getPath($deployment->getId() . '.' . \pathinfo('code.tar.gz', PATHINFO_EXTENSION));
+                $result = $deviceFunctions->write($path, $buffer, $mimeType);
```

Comment:
@Meldiron why are we replacing the usage of the `move` function with a read and write?

The `move` function is designed to handle chunking etc.

Reply:
Using move() does not support transferring between devices in the storage library version that we currently use. That support was introduced in https://github.com/utopia-php/storage/releases/tag/0.16.0

We will upgrade to the new storage library with Appwrite 1.5